packages/driver/src/cy/assertions.coffee

```diff
@@ -56,6 +58,27 @@ parseValueActualAndExpected = (value, actual, expected) ->
 
   obj
 
+prepareObjsForDiff = (err) ->
+  if _.isString(err.actual) || _.isString(err.expected)
+    return err
+  ret = {}
+  ret.actual = mochaUtils.stringify(err.actual)
+  ret.expected = mochaUtils.stringify(err.expected)
+  return ret
+
+objToString = Object.prototype.toString
+
+_sameType = (a, b) ->
+  return objToString.call(a) is objToString.call(b)
+
+showDiff = (err) ->
```

Comment:
this condition is too complex, AND and negations all mixed. I would suggest splitting this into little functions so that the meaning is clear from names

Reply:
I can't think of a clear way to reduce this down, what do you think? 

but I could rename the function to `shouldShowDiff` to make it clearer